src/votekit/election_types.py

```diff
@@ -336,7 +336,7 @@ def remove_cand(removed_cand: str, ballots: list[Ballot]) -> list[Ballot]:
     """
     Removes candidate from ranking of the ballots
     """
-    update = deepcopy(ballots)
+    update = pickle.loads(pickle.dumps(ballots, -1))
```

Comment:
`pickle` doesn't deepcopy, so this code isn't equivalent.  If you just need a copy here, use copy.copy instead as that'll be much faster than `pickle`.

[As a](https://docs.python.org/3/library/copy.html#copy.deepcopy)

Reply:
(Note: I haven't verified the code is correct without a `deepcopy`!)

Reply:
Oops, yeah — I had tested copy earlier and it wasn't passing the tests. I think maybe it's best to leave deepcopy for now and let someone other than me figure out how to optimize!